bool digital_out_0
bool digital_out_1
bool digital_out_2
bool digital_out_3
bool digital_out_4
bool digital_out_5
bool digital_out_6
bool digitla_out_7

---
bool done
